src/es/doramasyt/src/eu/kanade/tachiyomi/animeextension/es/doramasyt/Doramasyt.kt

```diff
@@ -38,145 +48,219 @@ class Doramasyt : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
-        Log.i("bruh ep", element.select("a").attr("href"))
-        val formatedEp = when {
-            epNum.isNotEmpty() -> epNum.toFloatOrNull() ?: 1F
-            else -> 1F
+    override fun searchAnimeParse(response: Response): AnimesPage {
+        val document = response.asJsoup()
+        val elements = document.select("div.col-lg-2.col-md-4.col-6 div.animes")
+        val nextPage = document.select("ul.pagination li:last-child a").any()
+        val animeList = elements.map { element ->
+            SAnime.create().apply {
+                setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+                title = element.select("div.animedtls p").text()
+                thumbnail_url = element.select("a img").attr("src")
+                description = element.select("div.animedtls p").text()
+            }
         }
-        episode.setUrlWithoutDomain(element.select("a").attr("href"))
-        episode.episode_number = formatedEp
-        episode.name = "Episodio $formatedEp"
+        return AnimesPage(animeList, nextPage)
+    }
 
-        return episode
+    override fun animeDetailsParse(response: Response): SAnime {
+        val document = response.asJsoup()
+        val anime = SAnime.create()
+        val sub = document.selectFirst(".herohead > .heroheadmain > h2")?.text()?.trim() ?: ""
+        val title = document.selectFirst("div.herohead div.heroheadmain h1")?.text()?.trim() ?: ""
+        anime.title = title + if (sub.isNotEmpty()) " ($sub)" else ""
+        anime.description = document.selectFirst("div.herohead div.heroheadmain div.flimdtls p.textComplete")!!.ownText()
+        anime.genre = document.select("div.herohead div.heroheadmain div.writersdiv div.nobel h6 a").joinToString { it.text() }
+        anime.status = parseStatus(document.select("div.herohead div.heroheadmain div.writersdiv div.state h6").text())
+        return anime
     }
 
-    private fun getNumberFromEpsString(epsStr: String): String {
-        return epsStr.filter { it.isDigit() }
+    override fun episodeListParse(response: Response): List<SEpisode> {
+        val document = response.asJsoup()
+        return document.select("div.mainrowdiv.pagesdiv div.jpage div.col-item").map { element ->
+            val epNum = element.select("a div.flimss div.dtlsflim p").text().filter { it.isDigit() }
+            val formatedEp = when {
+                epNum.isNotEmpty() -> epNum.toFloatOrNull() ?: 1F
+                else -> 1F
+            }
+            SEpisode.create().apply {
+                setUrlWithoutDomain(element.select("a").attr("abs:href"))
+                episode_number = formatedEp
+                name = "Episodio $formatedEp"
+            }
+        }.reversed()
     }
 
     override fun videoListParse(response: Response): List<Video> {
         val document = response.asJsoup()
         val videoList = mutableListOf<Video>()
         document.select("div.playermain ul.dropcaps li#play-video a.cap").forEach { players ->
```

Comment:
I think if you're using this many extractors, you should parallelize this loop.
Same for the other sources in this loop.